scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+
+        This function returns NaN when the argument is outside the domain.
+
+        >>> X.icdf([-0.1, 0, 1, 1.1])
+        array([ nan, -0.5,  0.5,  nan])
+
+        """
+        return self._icdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _icdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_icdf_formula'):
+            method = self._icdf_formula
+        elif self.tol is _null and self._overrides('_iccdf_formula'):
+            method = self._icdf_complementarity
+        else:
+            method = self._icdf_inversion
+        return method
+
+    def _icdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _icdf_complementarity(self, x, **kwargs):
+        return self._iccdf_dispatch(1 - x, **kwargs)
+
+    def _icdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._cdf_dispatch, x, kwargs=kwargs)
+
+    @_set_invalid_nan
+    def ilogccdf(self, x, *, method=None):
+        r"""Inverse of the log of the complementary cumulative distribution function
+
+        For more information about the inverse of the complementary cumulative
+        distribution function, see `ContinuousDistribution.icdf`.
+
+        Parameters
+        ----------
+        x : Array
+            The argument of the inverse of the logarithm of the complementary
+            cumulative distribution function (inverse log-CCDF).
+        method : {None, 'formula', 'complementarity', 'inversion'}
+            The strategy used to evaluate the inverse log-CCDF.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the inverse log-CCDF itself
+            - ``'complementarity'``: evaluate the inverse log-CDF at the
+                                     logarithmic complement of `x`.
+            - ``'inversion'``: solve numerically for the argument at which the
+                               log-CCDF is equal to `x`.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The inverse log-CCDF evaluated at the provided argument.
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`. The
+        inverse log-CCDF takes on its minimum value of :math:`l` at
+        :math:`x = \log(1) = 0` and its maximum value of :math:`r` at
+        :math:`x = \log(0) = -\infty`. Because the log-CCDF has range
+        :math:`[-\infty, 0]`, the inverse log-CDF is only defined on the
+        negative reals; for :math:`x > 0`, ``ilogcdf`` returns ``nan``.
+
+        Occasionally it is needed to find the argument of the CCDF for which
+        the resulting probability is very close to `0` or `1` - too close to
```

Comment:
For `0`, that's fitting. For `1`, that is not the term I'd use. So yes, intentionally described it rather than using a term that I would apply to only one of the two. If there are references that talk about, e.g. "underflowing to `1.0`", I can consider it. But that doesn't look right at first glance.

Reply:
Indeed, that does not sound right. So the general term is the loss of precision then.

Reply:
But this has to do with finite precision of the *input*, not "loss of precision" during the calculation.

One could say that `stats.norm.cdf(25)` suffers from "loss of precision" because the output is `1.0`, and that the solution is to use `stats.norm.logcdf(25)` (`-3.0566967063825604e-138`). 

This statement is about when you want to find a value (`25`) at which the CDF is very close to `1.0` but not equal to `1.0`. There is not a floating point representation of the number itself, but there is a floating point representation of the log of the number.

Reply:
@ev-br please LMK if I can resolve